### PR TITLE
[SPARK-21468][PYSPARK][ML] Python API for FeatureHasher

### PR DESCRIPTION
**Pull request by MLnick**
Add Python API for `FeatureHasher` transformer.

## How was this patch tested?

New doc test.

### CODE REVIEW

File: python/pyspark/ml/feature.py [hunk 1]
```diff
@@ -697,6 +698,82 @@ def getScalingVec(self):
...
+    @since("2.3.0")
+    def setParams(self, numFeatures=1 << 18, inputCols=None, outputCol=None):
+        """
+        setParams(self, numFeatures=1 << 18, inputCols=None, outputCol=None)
+        Sets params for this FeatureHasher.
+        """
+        kwargs = self._input_kwargs
+        return self._set(**kwargs)
+
+
```
Comment (BryanCutler) [inline python/pyspark/ml/feature.py]: Should there be a `getNumFeatures()` method to return the param?

**Review by BryanCutler — COMMENTED** [review]
LGTM!

File: python/pyspark/ml/feature.py (hunk 1 above)
Comment (BryanCutler) [inline python/pyspark/ml/feature.py]: Nevermind, I forgot it's in the shared param `HasNumFeatures`

**Comment by holdenk** [comment]
LGTM

**Comment by MLnick** [comment]
Merged to master. Thanks for the review!

**Pull request closed without merging**